fdb-extensions/src/main/java/com/apple/foundationdb/map/BunchedMap.java

```diff
@@ -140,6 +152,44 @@ public BunchedMap(@Nonnull BunchedSerializer<K,V> serializer, @Nonnull Comparato
         }
     }
 
+    /**
+     * Instrument a range read. The base implementation does nothing, but extenders are encouraged to
+     * override this method with their own implementations that, for example, records the total numbers
+     * of keys read and their sizes.
+     *
+     * @param readFuture a future that will complete to a list of keys and values
+     */
+    protected void instrumentRangeRead(@Nonnull CompletableFuture<List<KeyValue>> readFuture) {
```

Comment:
This should probably return a `CompletableFuture<List<KeyValue>>` as `timer.instrument` returns a future.

Reply:
Done